src/journal/journal-file.c

```diff
@@ -372,6 +407,13 @@ JournalFile* journal_file_close(JournalFile *f) {
         if (f->mmap && f->cache_fd)
                 mmap_cache_free_fd(f->mmap, f->cache_fd);
 
+        if (f->fd >= 0 && f->truncate_on_close) {
+                uint64_t p;
+
+                if (!journal_file_tail_end(f, &p))
```

Comment:
please use `>= 0` check here. A check with `!` kinda suggests this was a boolean we check here, but it really isn't. It's < 0 on failure and >= 0 on sucess...